azure-tests/src/main/java/fixtures/paging/AutoRestPagingTestServiceAsyncClient.java

```diff
@@ -0,0 +1,410 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package fixtures.paging;
+
+import com.azure.core.annotation.Generated;
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.exception.HttpResponseException;
+import com.azure.core.http.rest.PagedFlux;
+import fixtures.paging.implementation.PagingsImpl;
+import fixtures.paging.models.CustomParameterGroup;
+import fixtures.paging.models.PagingGetMultiplePagesLroOptions;
+import fixtures.paging.models.PagingGetMultiplePagesOptions;
+import fixtures.paging.models.PagingGetMultiplePagesWithOffsetOptions;
+import fixtures.paging.models.PagingGetOdataMultiplePagesOptions;
+import fixtures.paging.models.Product;
+
+/** Initializes a new instance of the asynchronous AutoRestPagingTestService type. */
+@ServiceClient(builder = AutoRestPagingTestServiceBuilder.class, isAsync = true)
+public final class AutoRestPagingTestServiceAsyncClient {
+    @Generated private final PagingsImpl serviceClient;
+
+    /**
+     * Initializes an instance of AutoRestPagingTestServiceAsyncClient class.
+     *
+     * @param serviceClient the service client implementation.
+     */
+    @Generated
+    AutoRestPagingTestServiceAsyncClient(PagingsImpl serviceClient) {
+        this.serviceClient = serviceClient;
+    }
+
+    /**
+     * A paging operation that must return result of the default 'value' node.
+     *
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the paginated response with {@link PagedFlux}.
+     */
+    @Generated
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<Product> getNoItemNamePages() {
+        return this.serviceClient.getNoItemNamePagesAsync();
+    }
+
+    /**
+     * A paging operation that must ignore any kind of nextLink, and stop after page 1.
+     *
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the paginated response with {@link PagedFlux}.
+     */
+    @Generated
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<Product> getNullNextLinkNamePages() {
+        return this.serviceClient.getNullNextLinkNamePagesAsync();
+    }
```

Comment:
without change in this PR, if `--context-client-method-parameter`, this class would be empty.